lib/media/ver20/set_video_encoder_configuration.ex

```diff
@@ -0,0 +1,99 @@
+defmodule Onvif.Media.Ver20.SetVideoEncoderConfiguration do
+  import SweetXml
+  import XmlBuilder
+
+  alias Onvif.Device
+  alias Onvif.Media.Ver20.Profile.VideoEncoder
+
+  def soap_action, do: "http://www.onvif.org/ver20/media/wsdl/SetVideoEncoderConfiguration"
+
+  @spec request(Device.t(), list) :: {:ok, any} | {:error, map()}
+  def request(device, args),
+    do: Onvif.Media.Ver20.Media.request(device, args, __MODULE__)
+
+  def request_body(%VideoEncoder{} = video_encoder_config) do
+    element(:"s:Body", [
+      element(:"tr2:SetVideoEncoderConfiguration", [
+        element(
+          :"tr2:Configuration",
+          %{
+            "token" => video_encoder_config.reference_token,
+            "profile" => video_encoder_config.profile
+          },
+          [
+            element(:"tt:Name", video_encoder_config.name),
+            element(:"tt:UseCount", video_encoder_config.use_count),
+            element(
+              :"tt:Encoding",
+              Keyword.fetch!(
+                Ecto.Enum.mappings(video_encoder_config.__struct__, :encoding),
+                video_encoder_config.encoding
+              )
+            ),
+            element(:"tt:Quality", trunc(video_encoder_config.quality)),
+            element(
+              :"tt:Resolution",
+              [
+                element(:"tt:Width", video_encoder_config.resolution.width),
+                element(:"tt:Height", video_encoder_config.resolution.height)
+              ]
+            ),
+            element(:"tt:RateControl", [
+              element(:"tt:FrameRateLimit", video_encoder_config.rate_control.frame_rate_limit),
+              element(:"tt:ConstantBitRate ", video_encoder_config.rate_control.constant_bitrate),
+              element(:"tt:BitrateLimit", video_encoder_config.rate_control.bitrate_limit)
+            ]),
+            multicast_element(video_encoder_config.multicast_configuration)
+          ]
+        )
+      ])
+    ])
+  end
+
+  defp multicast_element(%{ip_address: %{type: :ipv4}} = multicast_configuration) do
+    element(:"tt:Multicast", [
+      element(:"tt:Address", [
+        element(
+          :"tt:Type",
+          Keyword.fetch!(
+            Ecto.Enum.mappings(multicast_configuration.ip_address.__struct__, :type),
+            multicast_configuration.ip_address.type
+          )
+        ),
+        element(:"tt:IPv4Address", multicast_configuration.ipv4_address)
```

Comment:
as this is the only difference, it's a bit more readable to keep the entire element in the request body and have your pattern match as
```
defp ip_address_element(%{ip_address: %{type: :ipv4}} = multicast_configuration), do: element(:"tt:IPv4Address", multicast_configuration.ipv4_address)

defp ip_address_element(multicast_configuration), do: element(:"tt:IPv6Address", multicast_configuration.ip_address.ipv6_address)
```

Reply:
Done